Use a simple class to sanitize JDBC exceptions and also log them (#11843)
apache/druid: docs/configuration/index.md

When you specify an error response transform strategy other than `none`, Druid transforms the error responses from Druid services as follows:
 - For any query API that fails in the Router service, Druid sets the fields `errorClass` and `host` to null. Druid applies the transformation strategy to the `errorMessage` field.
 - For any SQL query API that fails, for example `POST /druid/v2/sql/...`, Druid sets the fields `errorClass` and `host` to null. Druid applies the transformation strategy to the `errorMessage` field.

>  - For any JDBC related exceptions, Druid will turn all checked exceptions into QueryInterruptedException otherwise druid will attempt to keep the exception to as the same type, unless original exception isn't owned by druid, Druid applies the transformation strategy to the `errorMessage` field.

Reviewer: `exception to as the same` -> `exception as the same`

>  - For any JDBC related exceptions, Druid will turn all checked exceptions into QueryInterruptedException otherwise druid will attempt to keep the exception to as the same type, unless original exception isn't owned by druid, Druid applies the transformation strategy to the `errorMessage` field.

Reviewer 2: ```suggestion
- Druid keeps all checked exceptions as the same exception type except for JDBC-related exceptions and exceptions that are not owned by Druid. Druid labels JDBC-related exceptions as `QueryInterruptedException`. For exceptions that are not originally owned by Druid, Druid applies the transformation strategy to the `errorMessage` field.
```
